catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PrePolicyPlugin.java

```diff
@@ -0,0 +1,94 @@
+package ddf.catalog.plugin;
+
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * The {@link PrePolicyPlugin} extension point allows for request/response processing before any
+ * policies are implemented at the very beginning of the plugin chain.
+ * <p>
+ * FOR INTERNAL USE ONLY
+ * </p>
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface PrePolicyPlugin {
+
+    /**
+     * @param input the {@link CreateRequest} to process
+     * @return the value of the processed {@link CreateRequest} to pass to the next {@link AccessPlugin}
+     * @throws StopProcessingException thrown to halt processing when a critical issue occurs during processing. This is
+     *                                 intended to prevent other plugins from processing as well.
+     */
+    CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException;
```

Comment:
It looks like you only added this plugin into the `CreateOperation`.  This plugin should be called from all relevant operations.